opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/BaseReferralNotificationDetailsPresenter.java

```diff
@@ -0,0 +1,34 @@
+package org.smartregister.chw.core.presenter;
+
+import org.smartregister.chw.core.contract.BaseReferralNotificationDetailsContract;
+import org.smartregister.chw.core.interactor.BaseReferralNotificationDetailsInteractor;
+
+import java.lang.ref.WeakReference;
+
+public class BaseReferralNotificationDetailsPresenter implements BaseReferralNotificationDetailsContract.Presenter {
+
+    private WeakReference<BaseReferralNotificationDetailsContract.View> view;
+    private BaseReferralNotificationDetailsContract.Interactor interactor;
+
+    public BaseReferralNotificationDetailsPresenter(BaseReferralNotificationDetailsContract.View view) {
+        this.view = new WeakReference<>(view);
+        interactor = new BaseReferralNotificationDetailsInteractor(this);
+    }
+
+    @Override
+    public void getReferralDetails(String referralTaskId, String notificationType) {
+        interactor.fetchReferralDetails(referralTaskId, notificationType);
+    }
+
+    @Override
+    public BaseReferralNotificationDetailsContract.View getView() {
+        if (view != null) {
+            return view.get();
+        }
+        return null;
+    }
+
+    public void setInteractor(BaseReferralNotificationDetailsContract.Interactor interactor) {
+        this.interactor = interactor;
```

Comment:
Isn't this handled by the constructor?

Reply:
This was added so I could mock the Interactor in test. Alternatively i could have allowed passing it as a constructor parameter but just adding the setter is enough for now to avoid further refactor.

Reply:
This was added to facilitate mocking of the interactor during test. The interactor is instantiated inside the constructor/ An alternative approach would be to pass the interactor as a parameter of the constructor but this would have required further refactoring whereas adding the setter would still suffice. Plus having a setter instead of passing it as a constructor parameter leaves the responsibility to the presenter.